### PR TITLE
Use juju-2.1 as a fallback.

### PR DESCRIPTION
**Pull request by ericsnowcurrently**
(fixes https://bugs.launchpad.net/landscape/+bug/1650644)

Currently "juju-2.0" is hard-coded as the binary name.  This change adds more flexibility and makes an attempt at using a binary that actually exists.

Note that this patch does more than the minimum to fix the bug.  This is because we will likely run into this problem again and again, so a slightly broader solution is warranted.  At the same time, an even broader solution is arguably valid here by the same reasoning.  However, that can be addressed separately if it's worth bothering.

Also note that this branch does not actually make any effort toward compatibility with Juju 2.1 (if any is required).  Instead it opens the door for such efforts, allowing us to run the landscape system tests using 2.1.

### CODE REVIEW

File: txjuju/cli.py [hunk 1]
```diff
@@ -17,6 +17,31 @@
...
+            _utils.Executable.find(juju)
+        except _utils.ExecutableNotFoundError:
+            continue
+        else:
+            return juju
+    else:
+        return supported[0]
+
+
+JUJU1 = _find_best_juju([
```
Comment (blackboxsw) [inline txjuju/cli.py]: Can we get a test that modifies path and validates that JUJU1 and JUJU2 variables are properly discovered in environments which don't have juju-1 or juju-2 or juju-2.0

File: txjuju/cli.py [hunk 2]
```diff
@@ -17,6 +17,31 @@
...
+        "juju-1",
+        "juju-1.25",
+        ])
+JUJU2 = _find_best_juju([
+        "juju-2",
+        "juju-2.0",  # the latest production release
+        # (lp:1650644) For now we must accommodate varying binary paths.
```
Comment (ericsnowcurrently) [inline txjuju/cli.py]: Doing it at module-level is the point.  It's a one-time cost to populate a module-level constant with several possible values.  The alternative is to do the lookup in Juju2CLI.__init__.  However, in that case the lookup cost would be applied to every place that creates a new Juju2CLI, even though the value will be the same every time.  This approach isn't considered poor form.

**Comment by ericsnowcurrently** [comment]
@blackboxsw, all your review comments have been addressed.

**Comment by blackboxsw** [comment]
Couple unit tests remain broken.
======================================================================
FAIL: test_juju1 (txjuju.tests.test___init__.GetCLIClassTests)
Passing in c.juju.JUJU1 should result in Juju1CLI.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/src/txjuju/txjuju/tests/test___init__.py", line 17, in test_juju1
    self.assertIs(cls, cli.Juju1CLI)
AssertionError: <class 'txjuju.cli.Juju1CLI'> is not <class 'txjuju.cli.Juju1CLI'>

======================================================================
FAIL: test_juju2 (txjuju.tests.test___init__.GetCLIClassTests)
Passing in c.juju.JUJU2 should result in Juju2CLI.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/src/txjuju/txjuju/tests/test___init__.py", line 22, in test_juju2
    self.assertIs(cls, cli.Juju2CLI)
AssertionError: <class 'txjuju.cli.Juju2CLI'> is not <class 'txjuju.cli.Juju2CLI'>

----------------------------------------------------------------------

**Comment by ericsnowcurrently** [comment]
@CanonicalLtd/landscape test

**Pull request merged**